week3/homework/src/todo.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const fs = require('fs');
+const uuid = require('uuid/v4');
+
+const DEFAULT_ENCODING = 'utf8';
+
+class Todo {
+  constructor(filename) {
+    this._filename = filename;
+  }
+
+  async create(description) {
+    const todos = await this.read();
+
+    const todo = {
+      id: uuid(),
+      done: false,
+
+      description,
+    };
+
+    todos.push(todo);
+
+    await this._save(todos);
+
+    return todo;
+  }
+
+  read() {
+    return new Promise(resolve => {
+      fs.readFile(this._filename, DEFAULT_ENCODING, (error, data) => {
+        if (error) return resolve([]);
```

Comment:
it turns out that there are other reasons why reading a file can fail, it is not just when the file is not there.
It would be best to explicitly check if the file exists instead of abusing the error like this.